### ISSUE TITLE
Can't export model to TensoRT(engine)

### ISSUE THREAD

**Issue by Jserax** [ultralytics/ultralytics#22735]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm trying export model to TensorRT with next code
```
from ultralytics import YOLO

model = YOLO("/home/grigorev.maksim57/projects/detection/best.pt")
model.export(format="engine", dynamic=True, workspace=16, nms=True, half=True, batch=16)
```
but i get error:
```
TensorRT: starting export with TensorRT 10.8.0.43...
[11/20/2025-08:53:52] [TRT] [I] [MemUsageChange] Init CUDA: CPU -2, GPU +0, now: CPU 1300, GPU 3063 (MiB)
[11/20/2025-08:53:55] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +2486, GPU +380, now: CPU 3779, GPU 3443 (MiB)
[11/20/2025-08:53:55] [TRT] [I] ----------------------------------------------------------------
[11/20/2025-08:53:55] [TRT] [I] Input filename:   /home/grigorev.maksim57/projects/detection/best.onnx
[11/20/2025-08:53:55] [TRT] [I] ONNX IR version:  0.0.9
[11/20/2025-08:53:55] [TRT] [I] Opset version:    19
[11/20/2025-08:53:55] [TRT] [I] Producer name:    pytorch
[11/20/2025-08:53:55] [TRT] [I] Producer version: 2.6.0
[11/20/2025-08:53:55] [TRT] [I] Domain:           
[11/20/2025-08:53:55] [TRT] [I] Model version:    0
[11/20/2025-08:53:55] [TRT] [I] Doc string:       
[11/20/2025-08:53:55] [TRT] [I] ----------------------------------------------------------------
TensorRT: input "images" with shape(-1, 3, -1, -1) DataType.FLOAT
TensorRT: output "output0" with shape(-1, 300, 6) DataType.FLOAT
TensorRT: building FP16 engine as /home/grigorev.maksim57/projects/detection/best.engine
[11/20/2025-08:53:55] [TRT] [I] Local timing cache in use. Profiling results in this builder pass will not be stored.
[11/20/2025-08:53:55] [TRT] [E] [shapeContext.cpp::checkVolume::3856] Error Code 2: Internal Error (Assertion interval.max() >= 0 failed. Condition::kVOLUME_BOUND is undefined for factor in interval [-1374389534720,-4])
ERROR ❌ TensorRT: export failure 24.8s: 'NoneType' object does not support the context manager protocol
```
if set ```workspace=None```  error:
```
11/20/2025-08:52:05] [TRT] [E] Error Code: 9: Skipping tactic 0x0000000000000000 due to exception [ir_builder.cpp:myelinGraphCompile:339] Stride order set illegally on intermediate tensor(s).
[11/20/2025-08:52:05] [TRT] [E] IBuilder::buildSerializedNetwork: Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[/Constant_62_output_0.../Slice_83]}.)
ERROR ❌ TensorRT: export failure 280.4s: 'NoneType' object does not support the context manager protocol
```
yolo checks:
```
Ultralytics 8.3.180 🚀 Python-3.12.2 torch-2.6.0+cu124 CUDA:0 (NVIDIA A800 80GB PCIe, 81156MiB)
Setup complete ✅ (48 CPUs, 1007.7 GB RAM, 1807.1/3575.1 GB disk)

OS                     Linux-6.1.0-35-amd64-x86_64-with-glibc2.35
Environment            Linux
Python                 3.12.2
Install                git
Path                   /home/grigorev.maksim57/projects/zaharov-classification/.venv/lib/python3.12/site-packages/ultralytics
RAM                    1007.75 GB
Disk                   1807.1/3575.1 GB
CPU                    AMD EPYC 9354 32-Core Processor
CPU count              48
GPU                    NVIDIA A800 80GB PCIe, 81156MiB
GPU count              1
CUDA                   12.4

numpy                  ✅ 1.26.4>=1.23.0
matplotlib             ✅ 3.10.0>=3.3.0
opencv-python          ✅ 4.11.0.86>=4.6.0
pillow                 ✅ 11.2.1>=7.1.2
pyyaml                 ✅ 6.0.2>=5.3.1
requests               ✅ 2.32.3>=2.23.0
scipy                  ✅ 1.15.3>=1.4.1
torch                  ✅ 2.6.0>=1.8.0
torch                  ✅ 2.6.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.21.0>=0.9.0
tqdm                   ✅ 4.67.1>=4.64.0
psutil                 ✅ 6.1.0
py-cpuinfo             ✅ 9.0.0
pandas                 ✅ 2.2.3>=1.1.4
ultralytics-thop       ✅ 2.0.14>=2.0.0
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Jserax, thank you for your detailed report and for using Ultralytics 🚀!  

This is an **automated response** to help move things forward quickly while an Ultralytics engineer reviews your issue and follows up soon 🙂  

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 **Bug Report** related to TensorRT export, please help us by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if possible. For example:
- A small, self-contained script showing the exact `YOLO()` load and `.export()` call you are using (you already shared a good start 👍)
- The exact model type (e.g., YOLO11n/s/m or custom) and whether the issue also occurs with a public checkpoint (such as `yolo11n.pt`)
- Any additional logs from earlier in the export process or ONNX export step, if available

If this is instead a custom training ❓ Question, please provide as much information as possible, including:
- Dataset description and a few example images
- Training command, hyperparameters, and logs
- Confirmation you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧  
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)  
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community  

## Upgrade

Before we dig deeper, please ensure you have upgraded to the latest `ultralytics` package (and that all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are satisfied) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This helps confirm whether the issue still occurs on the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If you can reproduce the TensorRT export issue in one of these reference environments with a public model and share the exact steps and logs, that will be extremely helpful for debugging 🙏

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Use latest Ultralytics
